src/OpenLoco/Entities/Misc.cpp

```diff
@@ -126,18 +126,87 @@ namespace OpenLoco
 
     static loco_global<int32_t, 0x112C876> _currentFontSpriteBase;
 
+    // 0x004FADD0
+    constexpr Map::Pos2 _wiggleAmounts[] = {
+        { 1, -1 },
+        { 1, 1 },
+        { -1, 1 },
+        { -1, -1 },
+    };
+
+    // clang-format off
+    // 0x004FAD21
+    constexpr int8_t _wiggleZAmounts[] = {
+        3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
+        2, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0,
+    };
+    // clang-format on
+
     // 0x0044063E
     void MoneyEffect::update()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
         if (getSubType() == MiscEntityType::windowCurrency)
         {
-            call(0x0044063E, regs);
+            invalidateSprite();
+            if (wiggle == 0)
+            {
+                wiggle = 21;
+            }
+            else
+            {
+                wiggle--;
+            }
+
+            if (var_26 > 160)
```

Comment:
`80h` -> 128?
![image](https://user-images.githubusercontent.com/7483209/153733880-5bc1ff4f-a090-415a-8fae-0fae5198625b.png)

Reply:
4FAC80 is an array that has an element at slot 160 that has the value 0x80. So i've reworked this so that its just ebp < 160

Reply:
This took me a stupid amount of time to comprehend, I'll leave my notes below as a fun reference, but I have one small question - `jz` will jump when the operands are equal, but you've got `>` here; shouldn't it be `==` or `>=`. I presume the > was just a safety in case the value goes above 160?

- Notes for my understanding
1. move MoneyEffect.var_26 into ebp
2. move _rotation into ebx
3. move 4FAC80[ebp] into ax, which is the same as move 4FAC80[MoneyEffect.var_26] into ax
4. compare al with 0x80. this is the same as compare 4FAC80[MoneyEffect.var_26] with 128
5. jz to 440806 - jz jumps if ZF is set, and ZF is only set when both operands of cmp are equal

So, we jump to 440806 if 4FAC80[MoneyEffect.var_26] == 0x80, and that will be true if var_26 == 160

Reply:
Yeah i went for > just as a safety

Reply:
Shouldn't it still be `>=` though? jz compares equality, so just having `>` will be an off-by-one here